modules/datadog/variables.tf

```diff
@@ -63,3 +63,8 @@ variable "datadog_agent_version_fargate" {
   type        = string
   default     = "7.54.0"
 }
+
+variable "product_name" {
```

Comment:
I would prefer team_name, but I put product to align with the currently used tag in datadog (e.g. in disco pmd sfmc titan etc we use product:<teamname> so it felt more accurate calling the var product as well) 
I don't have a strong opinion tho